### ISSUE TITLE
Aria-labelledby should precede aria-label attribute

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#1283]
**Reported by simekk on 2010-12-18 18:45**
As we have it now, aria-label overrides aria-labelledby, but according to the specification, it should be the other way around.
(http://www.w3.org/TR/wai-aria/complete#textalternativecomputation)

EXAMPLE:
<input type="checkbox" id="flash" aria-label="this should not be used" aria-labelledby="this should be used">

OUTPUT:
this should not be used checkbox checked

Tested under IE8 and latest FF.

**Comment by dkager** [comment]
Sounds like a good one for the 2017 browse mode improvement project whose name I conveniently forgot.

**Comment by bhavyashah** [comment]
Project Webfix, perhaps?

On 7/2/17, Davy Kager <notifications@github.com> wrote:
> Sounds like a good one for the 2017 browse mode improvement project whose
> name I conveniently forgot.
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/1283#issuecomment-312488609


-- 
Best Regards
Bhavya Shah

Blogger at Hiking Across Horizons: https://bhavyashah125.wordpress.com/

Contacting Me
E-mail Address: bhavya.shah125@gmail.com
Follow me on Twitter @BhavyaShah125 or www.twitter.com/BhavyaShah125
Mobile Number: +91 7506221750

**Comment by dkager** [comment]
@bhavyashah Yes.

**Comment by jcsteh** [comment]
I don't think this is a candidate for webfix:

1. For any browser other than IE, this is determined by the browser, not by NVDA. So, the fixes need to be made in the browsers.
2. IE specific fixes are low priority for us than other browsers.
3. Having both aria-labelledby and aria-label seems pretty obscure and unlikely, even in the wild. Webfix is really about issues that hurt quite a number of users in the wild, not spec compliance.

Giving this p4, since I honestly don't see this hurting real use cases. If someone can present a use case where this really matters for IE usage, we might consider revising this to p3.

**Comment by Adriani90** [comment]
@simekk The link in the description is not working anymore. Could you please upload a new test case? Thanks!

**Comment by fstrr** [comment]
I put a [test case](https://cdpn.io/pen/debug/rNdZmKX?authentication_hash=yPAJjbyEyZPA) together for this. The issue has been fixed now, so this can be closed.

**Comment by seanbudd** [comment]
Great, thanks @fstrr
